### PR TITLE
Hide Angular renderer hints by default #1576

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
- set and update config
- use default config when not set
- show/hide renderer hints based on config

### CODE REVIEW

File: packages/angular/src/abstract-control.ts [hunk 1]
```diff
@@ -86,6 +86,14 @@ export abstract class JsonFormsAbstractControl<
     this.triggerValidation();
   }
 
+  shouldShowUnfocusedDescription(): boolean {
+    const config = this.jsonFormsService.getState().jsonforms.config;
```
Comment (sdirix) [inline packages/angular/src/abstract-control.ts:90]: The `config` is just an easy way to set default options for ui schema elements. If the same option exists on a ui schema element it has a higher precedence then the one from the config.

**Comment by sdirix** [comment]
I'll take care of the fix.

**Comment by AlexandraBuzila** [comment]
I rebased the changes the fix the conflicts with master.

File: packages/angular/src/abstract-control.ts [hunk 2]
```diff
@@ -86,6 +87,15 @@ export abstract class JsonFormsAbstractControl<
...
+    const appliedUiSchemaOptions = merge({}, config, this.uischema.options);
+    if (appliedUiSchemaOptions) {
+      return !!appliedUiSchemaOptions.showUnfocusedDescription;
+    }
+    return false;
```
Comment (sdirix) [inline packages/angular/src/abstract-control.ts:96]: Here we hide away the default value for `showUnfocusedDescription` in this method. I would prefer when the service would return the default `config` when the user did not set their own. This way we also don't need the `undefined` check and can just immediately return the value of the config.

Comment (AlexandraBuzila) [reply packages/angular/src/abstract-control.ts:96]: The config here already contains the default values for undefined properties (the default values are set by the configReducer https://github.com/eclipsesource/jsonforms/pull/1584/files#diff-42788974ca6b5be621ff5d539f3155e8R83). I guess I added the undefined checks out of habit, would you prefer if I removed them?

Comment (sdirix) [reply packages/angular/src/abstract-control.ts:96]: Maybe I'm missing something but isn't the config `undefined` by default and will only be set when the user actually calls `updateConfig`? Ideally it should always set to its default state when the user did not call `updateConfig`.

Comment (AlexandraBuzila) [reply packages/angular/src/abstract-control.ts]: You're right, I see your point now. The last comit fixes this.

**Pull request merged**